modules/datahandling/filerequests.go

```diff
@@ -309,9 +318,12 @@ func (f fileChangeRequest) process(db dbfs.DBFS) ([]dhClosure, error) {
 			"SenderID":  f.SenderID,
 			"ProjectID": fileMeta.ProjectID,
 		})
+		ack()
 		return []dhClosure{toSenderClosure{msg: messages.NewEmptyResponse(messages.StatusUnauthorized, f.Tag)}}, nil
 	}
 
+	ack() // ack just before appending so if it fails in the middle it doesn't re-enqueue it
```

Comment:
I think we should still ack after, since CPAppendFileChange returns an error for CAS mismatches, doesn't it?